pymc4/mcmc/utils.py

```diff
@@ -127,12 +129,17 @@ def trace_to_arviz(
             return trace + az.from_dict(posterior_predictive=posterior_predictive)
         else:
             trace = None
+    if log_likelihood is not None and isinstance(log_likelihood, dict):
+        log_likelihood = {
+            k: np.swapaxes(v.numpy(), 1, 0) for k, v in log_likelihood.items() if "/" in k
+        }
 
     return az.from_dict(
         posterior=trace,
         sample_stats=sample_stats,
         prior_predictive=prior_predictive,
         posterior_predictive=posterior_predictive,
+        log_likelihood=log_likelihood,
         observed_data=observed_data,
     )
```

Comment:
I am a bit confused here whether log_likelihood has to be a group or should it be a variable in `sample_stats`.

Reply:
As per discussions in ArviZ gitter channel, it's better to have `log_likelihood` as a group for compatibility with Multi-observed random variables. So, I will not change `trace_to_arviz` function.